middlewares/validators/user.js

```diff
@@ -68,10 +68,11 @@ const validateJoinData = async (req, res, next) => {
       funFact: joi.string().min(100).required(),
       whyRds: joi.string().min(100).required(),
       flowState: joi.string().optional(),
+      numberOfHours: joi.number().min(1).max(100).required(),
     });
 
   try {
-    await schema.validateAsync(req.body);
+    await schema.validateAsync({ ...req.body, numberOfHours: parseInt(req.body.numberOfHours) });
```

Comment:
Why are we passing numberofHours here?
And why are we parsing it here?

Reply:
The expected value of numberOfHours is an integer. If i dont parse it, it get saved us string in firebase db.

Reply:
The reason why are we parsing it there, is bcz at that line it gets saved to db

Reply:
I don't get it, Won't joi throw an error if it's not a number, why parse and send?

Reply:
Yes, so the problem:
Req.body by default will be a string.
If i make it string in joi, i wont be able to apply min and max to it, so the user can send invalid data still. And now if i make it number, and validate the schema, it will throw error, so i made numberOfHours in req.body a number to work around it.

This way, i can have my validation for the feild having number between 1 to 100, and also make sure that the data being sent to db is in correct form. 

I hope my point makes sense 😅
Please tell an alternative way, if there is one.